File: src/main/java/br/org/acal/resouces/model/LinkModel.java
package br.org.acal.resouces.model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.io.Serializable;

@Entity
@Table(name = "enderecopessoa")
@Data
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class LinkModel implements Serializable {

    @Id
    @Column(name = "id")
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private String number;

    @JoinColumn( name = "idPessoa", referencedColumnName = "id")
    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    private CustomerModel customer;

    @JoinColumn(name = "idEndereco", referencedColumnName = "id")
    @ManyToOne(fetch = FetchType.LAZY,optional = false)
    private AddressModel address;

    @JoinColumn(name = "idCategoriaSocio", referencedColumnName = "id")
    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    private CategoryModel category;

    @Column(name = "Numero")
    private String linkNumber;

    @Column(name = "socioExclusivo")
    private Boolean exclusiveMember;

    @Column(name = "inativo")
    private Boolean inactive;

}

File: src/main/kotlin/br/org/acal/resources/model/LinkModel.kt
package br.org.acal.resources.model

import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.Id
import jakarta.persistence.JoinColumn
import jakarta.persistence.ManyToOne
import jakarta.persistence.Table
import java.io.Serializable
import lombok.AllArgsConstructor
import lombok.Builder
import lombok.Data
import lombok.NoArgsConstructor

@Entity
@Table(name = "enderecopessoa")
@Data
@AllArgsConstructor
@NoArgsConstructor
@Builder
class LinkModel : Serializable {
    @Id
    @Column(name = "id")
    private var number: String? = null

    @JoinColumn(name = "idPessoa", referencedColumnName = "id")
    @ManyToOne(optional = false)
    private val customer: CustomerModel? = null

    @JoinColumn(name = "idEndereco", referencedColumnName = "id")
    @ManyToOne(optional = false)
    private val address: AddressModel? = null

    @JoinColumn(name = "idCategoriaSocio", referencedColumnName = "id")
    @ManyToOne(optional = false)
    private val category: CategoryModel? = null

    @Column(name = "Numero")
    private var partnerNumber: String? = null
}
